lib/Basics/Endian.h

```diff
@@ -46,197 +46,121 @@ inline bool isLittleEndian(){
   return rv;
```

Comment:
can we avoid the static variable here altogether? whenever this function is entered, there will be an implicit exclusive lock on the static variable's initialization? can't we initialize the static variable outside of a function?

Reply:
we could easily, if we did the preprocessing differently. we should provide an #else for the  #ifdef, which errors.